pallets/rewards/src/lib.rs

```diff
@@ -206,26 +214,29 @@ pub mod pallet {
 	impl<T: Config<I>, I: 'static> GroupRewards for Pallet<T, I>
 	where
 		RewardGroupOf<T, I>: FullCodec + Default,
+		DistributionIdOf<T, I>: FullCodec + Default,
 	{
 		type Balance = BalanceOf<T, I>;
 		type GroupId = T::GroupId;
 
 		fn reward_group(group_id: Self::GroupId, reward: Self::Balance) -> DispatchResult {
-			Groups::<T, I>::try_mutate(group_id, |group| {
-				T::RewardMechanism::reward_group(group, reward)?;
+			LastDistributionId::<T, I>::try_mutate(|distribution_id| {
+				Groups::<T, I>::try_mutate(group_id, |group| {
+					T::RewardMechanism::reward_group(group, reward, distribution_id.next_id()?)?;
 
-				T::Currency::mint_into(
-					T::RewardCurrency::get(),
-					&T::PalletId::get().into_account_truncating(),
-					reward,
-				)?;
+					T::Currency::mint_into(
+						T::RewardCurrency::get(),
+						&T::PalletId::get().into_account_truncating(),
```

Comment:
I like the idea. Maybe we can reuse the idea of Frederic about a configurable minting method and specify that there. I add this to https://github.com/centrifuge/centrifuge-chain/issues/1056 to track it as another PR in case we decide to modify this